src/Core/Generator/SamplingModes.cs

```diff
@@ -12,18 +12,18 @@ public enum SamplingModes
         /// Represents the mode where the top N records are sampled.
         /// This mode selects a specified number of records in descending order based on a timestamp or other sorting criteria.
         /// </summary>
-        TopNSampler,
+        TopNExtractor,
 
         /// <summary>
         /// Represents the mode where data is sampled based on partitions.
         /// In this mode, data is fetched from each partition with a specified limit per partition, which helps in distributing the sampling across different data segments.
         /// </summary>
-        PartitionBasedSampler,
+        EligibleDataSampler,
```

Comment:
Where does the name "eligibledata" come from? all the descriptions and design talk about "partition" but i don't see the relation to "eligibledata"

Reply:
Behind the scene, we are using partition information to collect the sample data but for customer, we are not asking for any partition related input parameter, so name might confuse them, Hence, renamed to this. @sajeetharan  do you want to add here?

Reply:
gotcha, thanks for clarifying